reactioncommerce/reaction-next-starterkit
# feat: use CatalogGrid from component library

Resolves #242 and 4543 from the main reaction repo
Impact: **minor**
Type: **feature**

## Solution
This PR ports back the new `CatalogGrid` component from the component library to Starterkit. The functionality is mostly the same, with a few exceptions:

- Grid is responsive to its container. Depending on container width, 1, 2, 3 or 4 products will display per row
- Images inside the grid (`ProgressiveImage`) are also responsive to their container. This allows for more accurate scaling.

## Breaking changes
None

## Testing
1. Run reaction with this PR: https://github.com/reactioncommerce/reaction/pull/4565
2. View the grid and confirm it looks and operates correctly
3. Confirm you can click through to the PDP and that tracking works. Add logging to lib/tracking/trackProductClicked.js so that you can inspect the data before it is returned.
4. Click through the app and confirm images are still working everywhere - grid, pdp, cart popout and page.
